lib/Elevate/Leapp.pm

```diff
@@ -233,6 +236,66 @@ sub extract_error_block_from_output ( $self, $text_ar ) {
     return $error_block;
 }
 
+sub check_upgrade_log_for_failures ($self) {
+    my $upgrade_log = LEAPP_UPGRADE_LOG;
+    my $cont_file   = LEAPP_FAIL_CONT_FILE;
+    my $max_delay   = 60 * 10;                # 10 minutes. If changed don't forget to update the INFO line below.
+
+    if ( !-e $upgrade_log ) {
+        ERROR("No LEAPP upgrade log detected at [$upgrade_log]");
+
+        return 1;
+    }
+
+    my $contents;
+    my $not_found = 1;
+    my $elapsed   = 0;
+    INFO("Waiting to ensure the LEAPP upgrade process completes. This process may take up to 10 minutes.");
+    my $found = $self->_wait_for_log_contents( 'Starting stage After of phase FirstBoot', $max_delay );
+
+    if ( !$found ) {
+        if ( -e $cont_file ) {
+            INFO("LEAPP does not appear to have completed successfully, but continuing anyway because [$cont_file] exists");
+
+            return 0;
+        }
+        else {
+            my $msg = sprintf("The command 'leapp upgrade' did not complete successfully. Please investigate and resolve the issue.\n");
+            $msg .= sprintf("Once resolved, you can continue the upgrade by running the following commands:\n");
+            $msg .= sprintf( "    %s\n", "touch $cont_file" );
+            $msg .= sprintf( "    %s\n", '/scripts/elevate-cpanel --continue' );
+            $msg .= sprintf("The following log files may help in your investigation.\n");
+            $msg .= sprintf( "    %s\n", $upgrade_log );
+            $msg .= sprintf( "    %s\n", LEAPP_REPORT_TXT );
+            ERROR($msg);
+
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+sub _wait_for_log_contents ( $self, $search_string, $max_delay ) {
+    my $upgrade_log = LEAPP_UPGRADE_LOG;
+    my $found       = 0;
+    my $elapsed     = 0;
+
+    DEBUG("Waiting for the LEAPP upgrade log to contain [$search_string]. This process may take up to $max_delay seconds.");
```

Comment:
That question is why I had this as a debug. I meant this to really only be for our use, but I can absolutely change it if you think it would be more useful to have this in addition to the other stuff that gets printed out.

The log looks like this with the current changes:
```
* 2024-04-05 22:27:18 (7768) [INFO] ******************************************************************************************
* 2024-04-05 22:27:18 (7769) [INFO] *
* 2024-04-05 22:27:18 (7770) [INFO] * Starting stage 4 of 5
* 2024-04-05 22:27:18 (7771) [INFO] *
* 2024-04-05 22:27:18 (7772) [INFO] ******************************************************************************************
* 2024-04-05 22:27:18 (5172) [INFO] Waiting to ensure the LEAPP upgrade process completes. This process may take up to 10 minutes.
* 2024-04-05 22:27:18 (5206) [DEBUG] Waiting for the LEAPP upgrade log to contain [Starting stage After of phase FirstBoot]. This process may take up to 600 seconds.
* 2024-04-05 22:27:43 (7717) [DEBUG] Running 'post_leapp' from RpmDB component.
* 2024-04-05 22:27:43 (5681) [INFO] Running: /usr/bin/dnf clean all
```

With that all, if you still think it should be changed, I'll change it, just want to make sure is all

Reply:
oh, I see.  I think it is fine to leave this line as it is or even remove it in that case.